sw/host/ot_certs/src/cwt.rs

```diff
@@ -0,0 +1,354 @@
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Result, Context};
+use heck::{ToShoutySnakeCase, ToUpperCamelCase};
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+use crate::codegen::Codegen;
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct CwtTemplate {
+    /// Name of the certificate.
+    pub name: String,
+    /// Variable declarations.
+    pub variables: IndexMap<String, VariableType>,
+    /// Variable declarations.
+    pub constants: IndexMap<String, ConstantType>,
+    /// Certificate specification.
+    pub certificate: StructureType,
+}
+
+/// Declaration of a variable that can be filled into the template.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(tag = "type", rename_all = "kebab-case")]
+pub enum VariableType {
+    /// Raw array of bytes.
+    ByteArray {
+        /// Length in bytes for this variable.
+        size: usize,
+    },
+    /// Integer.
+    Integer,
+    /// UTF-8 encoded String.
+    String {
+        /// Maximum size in bytes for this variable.
+        size: usize,
+    },
+}
+
+impl VariableType {
+    fn get_max_size(&self) -> usize {
```

Comment:
document what this is the maximum size of (ie the total encoding, not just the data)